src/environment.cc

```diff
@@ -179,6 +180,29 @@ const std::string& Environment::KubernetesClusterName() const {
   return kubernetes_cluster_name_;
 }
 
+const std::string& Environment::KubernetesClusterLocation() const {
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (kubernetes_cluster_location_.empty()) {
+    if (!config_.KubernetesClusterLocation().empty()) {
+      kubernetes_cluster_location_ = config_.KubernetesClusterLocation();
+    } else {
+      // Get the kube-env.
+      const std::string kube_env =
+          GetMetadataString("instance/attributes/kube-env");
+      // kube-env is a list of NAME: VALUE pairs, one per line.
+      // The actual location is in the ZONE variable.
```

Comment:
Is this true for regional clusters too? Or is there a new region / location field?

Reply:
This was stale anyway. See the new shiny in the next push.